### ISSUE TITLE
KC_DB_USERNAME and KC_DB_PASSWORD are not respected when properties are set

### ISSUE THREAD

**Issue by dev-e** [keycloak/keycloak#12413]
### Describe the bug

When `db-username` and `db-password` properties are defined in the `/opt/keycloak/conf/keycloak.conf` file and `KC_DB_USERNAME` and `KC_DB_PASSWORD` environment variables are set, Keycloak does not respect these environment variables and uses values from the file. However args `--db-username` and `--db-password` override properties from the file.

### Version

18.0.0

### Expected behavior

`KC_DB_USERNAME` and `KC_DB_PASSWORD` environment variables should override `db-username` and `db-password` properties defined in the `/opt/keycloak/conf/keycloak.conf` file

### Actual behavior

`db-username` and `db-password` properties are used instead of respectful environment variables.

### How to Reproduce?

Dockerfile:
```
ARG BASE_IMAGE_PATH
ARG BASE_IMAGE_VERSION

FROM ${BASE_IMAGE_PATH}:${BASE_IMAGE_VERSION}

ENV KC_DB=postgres

COPY conf/ /opt/keycloak/conf/
RUN /opt/keycloak/bin/kc.sh build

```
conf/keycloak.conf file:
```
# Database settings
db=postgres
db-url-host=postgres
db-url-port=5432
db-url-database=keycloak
db-schema=public
db-username=keycloak
db-password=password

# Cache settings
cache=local

# Hostname settings
hostname-strict=false

# HTTP settings
http-enabled=true
http-relative-path=auth

# Proxy mode
proxy=edge

# Metrics and healthcheck settings
metrics-enabled=true
health-enabled=true

# Enabled features
features.admin_fine_grained_authz=enabled
features.client_secret_rotation=enabled
features.token_exchange=enabled

# Logging configuration
log-level=info
log-console-color=true

# Override for local development mode
%dev.spi-theme-cache-themes=false
%dev.spi-theme-cache-templates=false
%dev.spi-theme-static-max-age=-1

# Logging configuration. INFO is the default level for most of the categories
#quarkus.log.level = DEBUG
quarkus.log.category."org.jboss.resteasy.resteasy_jaxrs.i18n".level=WARN
quarkus.log.category."org.infinispan.transaction.lookup.JBossStandaloneJTAManagerLookup".level=WARN
quarkus.log.sentry=true
quarkus.log.sentry.dsn=https://sentry
quarkus.log.sentry.in-app-packages=*
quarkus.log.sentry.level=WARN
quarkus.log.sentry.server-name=keycloak
quarkus.log.sentry.release=18.0.0-iponweb-10.0
quarkus.log.sentry.environment=dev
```
Build an image:
```
docker build --build-arg BASE_IMAGE_PATH=quay.io/keycloak/keycloak --build-arg BASE_IMAGE_VERSION=18.0.0 . -t keycloak_test
```
Run the container with valid DB credentials:
```
docker run --rm -e KC_DB_USERNAME=keycloak_test -e KC_DB_PASSWORD="<PASSWORD>" -e KC_DB_URL=jdbc:postgresql://localhost:5432/keycloak_x_test keycloak_test --verbose start
```
Logs:
```
2022-06-08 22:24:30,936 WARN  [org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator] (JPA Startup Thread: keycloak-default) HHH000342: Could not obtain connection to query metadata: org.postgresql.util.PSQLException: FATAL: password authentication failed for user "keycloak"
	at org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:646)
	at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:180)
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:235)
	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:49)
	at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:223)
	at org.postgresql.Driver.makeConnection(Driver.java:400)
	at org.postgresql.Driver.connect(Driver.java:259)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:677)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:228)
	at org.postgresql.ds.common.BaseDataSource.getConnection(BaseDataSource.java:103)
	at org.postgresql.xa.PGXADataSource.getXAConnection(PGXADataSource.java:49)
	at org.postgresql.xa.PGXADataSource.getXAConnection(PGXADataSource.java:35)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:216)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:513)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:494)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1126)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

2022-06-08 22:24:31,152 INFO  [org.infinispan.CONTAINER] (keycloak-cache-init) ISPN000128: Infinispan version: Infinispan 'Triskaidekaphobia' 13.0.8.Final
2022-06-08 22:24:33,646 WARN  [io.agroal.pool] (agroal-11) Datasource '<default>': FATAL: password authentication failed for user "keycloak"
2022-06-08 22:24:33,934 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to start server in (production) mode
```

**Comment by dev-e** [comment]
By the way, the same is applicable for `log-level` option. Being specified in the `keycloak.conf` file it is not overridden with the `KC_LOG_LEVEL` environment variable.

**Comment by DGuhr** [comment]
@dev-e thanks for catching this one! @pedroigor provided a fix in #12638
